tests/FSharp.Test.Utilities/XunitHelpers.fs

```diff
@@ -145,26 +145,48 @@ type FSharpXunitFramework(sink: IMessageSink) =
                 // We need AssemblyResolver already here, because OpenTelemetry loads some assemblies dynamically.
                 AssemblyResolver.addResolver ()
             #endif
+
+                // Override cache capacity to reduce memory usage in CI.
+                FSharp.Compiler.Cache.OverrideMaxCapacityForTesting()
+
+                let testRunName = $"RunTests_{assemblyName.Name} {Runtime.InteropServices.RuntimeInformation.FrameworkDescription}"
+                
+                // On Windows forwarding localhost to wsl2 docker container sometimes does not work. Use IP address instead.
+                let otlpEndpoint = Uri("http://127.0.0.1:4317")
                 
                 // Configure OpenTelemetry export. Traces can be viewed in Jaeger or other compatible tools.
                 use tracerProvider =
                     OpenTelemetry.Sdk.CreateTracerProviderBuilder()
                         .AddSource(ActivityNames.FscSourceName)
                         .ConfigureResource(fun r -> r.AddService("F#") |> ignore)
                         .AddOtlpExporter(fun o ->
+                            o.Endpoint <- otlpEndpoint
+                            o.Protocol <- OpenTelemetry.Exporter.OtlpExportProtocol.Grpc
                             // Empirical values to ensure no traces are lost and no significant delay at the end of test run.
                             o.TimeoutMilliseconds <- 200
                             o.BatchExportProcessorOptions.MaxQueueSize <- 16384
                             o.BatchExportProcessorOptions.ScheduledDelayMilliseconds <- 100
                         )
                         .Build()
 
+                use meterProvider =
+                    OpenTelemetry.Sdk.CreateMeterProviderBuilder()
+                        .AddMeter(nameof FSharp.Compiler.CacheInstrumentation)
+                        .AddMeter("System.Runtime")
```

Comment:
This "System.Runtime" meter works only for net9.0 but is actually quite awesome. GC, memory, and lots of other metrics can be viewed in Prometheus in real time while the tests run. I use a simple [docker setup like this](https://github.com/[majocha/otel](https://github.com/majocha/otel))